src/js/angular/ttyg/controllers/ttyg-view.controller.js

```diff
@@ -313,6 +250,29 @@ function TTYGViewCtrl($rootScope, $scope, $http, $timeout, $translate, $uibModal
         }
     };
 
+    const onCreateNewChat = (chatQuestion) => {
+        let newConversationId = undefined;
+        TTYGService.createConversation($scope.chatQuestion)
+            .then((conversationId) => {
+                newConversationId = conversationId;
+                return loadChats();
+            })
+            .then(() => {
+                TTYGContextService.emit(TTYGEventName.CREATE_CHAT_SUCCESSFUL);
+                TTYGContextService.selectChat(TTYGContextService.getChats().getChat(newConversationId));
+            })
+            .catch((error) => {
+                TTYGContextService.emit(TTYGEventName.CREATE_CHAT_FAILURE);
+                toastr.error($translate.instant('ttyg.chat.messages.create_failure'));
+            });
+    };
+
+    const onAskQuestion = (chatQuestion) => {
+        TTYGService.askQuestion(chatQuestion)
+            .then((answer) => TTYGContextService.emit(TTYGEventName.ASK_QUESTION_SUCCESSFUL, answer))
+            .catch((error) => toastr.error(getError(error, 0, 100)));
```

Comment:
We should add some progress indicator somewhere during this request, because it turned out to be quite slow at most of the time.

Reply:
This will be done in a future request.